server.go

```diff
@@ -313,9 +313,9 @@ func (sp *serverPeer) addKnownAddresses(addresses []*wire.NetAddress) {
 
 // addressKnown true if the given address is already known to the peer.
 func (sp *serverPeer) addressKnown(na *wire.NetAddress) bool {
-	sp.addrMtx.Lock()
+	sp.addrMtx.RLock()
+	defer sp.addrMtx.Unlock()
```

Comment:
Glad it was helpful.

Regarding the RWMutex, I haven't used RWMutxex in go before, but it looks like the R/W Mutexes work on different semaphores? I tried this code and got a panic:

```
	rwMtx := sync.RWMutex{}
	rwMtx.RLock()
	// rwMtx.RUnlock()
	rwMtx.Unlock()
```
I think you need to use `RUnlock()`?